crd-diff-config.yaml

```diff
@@ -1,109 +1,5 @@
-    sameVersion:
-      enabled: true
-      unhandledFailureMode: "Closed"
-      enum:
-        enabled: true
-        removalEnforcement: "Strict"
-        additionEnforcement: "Strict"
-      default:
-        enabled: true
-        changeEnforcement: "Strict"
-        removalEnforcement: "Strict"
-        additionEnforcement: "Strict"
-      required:
-        enabled: true
-        newEnforcement: "Strict"
-      type:
-        enabled: true
-        changeEnforcement: "Strict"
-      maximum:
-        enabled: true
-        additionEnforcement: "Strict"
-        decreaseEnforcement: "Strict"
-      maxItems:
-        enabled: true
-        additionEnforcement: "Strict"
-        decreaseEnforcement: "Strict"
-      maxProperties:
-        enabled: true
-        additionEnforcement: "Strict"
-        decreaseEnforcement: "Strict"
-      maxLength:
-        enabled: true
-        additionEnforcement: "Strict"
-        decreaseEnforcement: "Strict"
-      minimum:
-        enabled: true
-        additionEnforcement: "Strict"
-        increaseEnforcement: "Strict"
-      minItems:
-        enabled: true
-        additionEnforcement: "Strict"
-        increaseEnforcement: "Strict"
-      minProperties:
-        enabled: true
-        additionEnforcement: "Strict"
-        increaseEnforcement: "Strict"
-      minLength:
-        enabled: true
-        additionEnforcement: "Strict"
-        increaseEnforcement: "Strict"
-    servedVersion:
-      enabled: true
-      unhandledFailureMode: "Closed"
-      enum:
-        enabled: true
-        removalEnforcement: "Strict"
-        additionEnforcement: "Strict"
-      default:
-        enabled: true
-        changeEnforcement: "Strict"
-        removalEnforcement: "Strict"
-        additionEnforcement: "Strict"
-      required:
-        enabled: true
-        newEnforcement: "Strict"
-      type:
-        enabled: true
-        changeEnforcement: "Strict"
-      maximum:
-        enabled: true
-        additionEnforcement: "Strict"
-        decreaseEnforcement: "Strict"
-      maxItems:
-        enabled: true
-        additionEnforcement: "Strict"
-        decreaseEnforcement: "Strict"
-      maxProperties:
-        enabled: true
-        additionEnforcement: "Strict"
-        decreaseEnforcement: "Strict"
-      maxLength:
-        enabled: true
-        additionEnforcement: "Strict"
-        decreaseEnforcement: "Strict"
-      minimum:
-        enabled: true
-        additionEnforcement: "Strict"
-        increaseEnforcement: "Strict"
-      minItems:
-        enabled: true
-        additionEnforcement: "Strict"
-        increaseEnforcement: "Strict"
-      minProperties:
-        enabled: true
-        additionEnforcement: "Strict"
-        increaseEnforcement: "Strict"
-      minLength:
-        enabled: true
-        additionEnforcement: "Strict"
-        increaseEnforcement: "Strict"
+validations:
+  - name: enum
+    enforcement: Error
+    configuration:
+      additionPolicy: Allow
```

Comment:
Note for reviewers: This is technically a less restrictive config than existed before, but IIRC it was planned to add new enum values in various places during continued development of OLMv1 so this felt reasonable to include.

If you want to maintain the strictest possible configuration, the options here could be to:
- Remove the use of a configuration file as `crdify` defaults to the strictest possible analysis
- Switch `additionPolicy` to `Disallow` (keeps the config file around)